### PR TITLE
Html whitelist

### PR DESCRIPTION
**Pull request by anitagraham**
This is, I think, complete.
The tests work, and test only the add_whitelist features.
The code includes all of @bricesanchez changes.

If there are git knots I gnot up to untangling them. :smile:

### CODE REVIEW

File: pages/lib/refinery/pages/configuration.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require 'set'
```
Comment (parndt) [inline pages/lib/refinery/pages/configuration.rb]: where is `Set` used?

Comment (bricesanchez) [inline pages/lib/refinery/pages/configuration.rb]: I've forgotten to remove it, thanks for your review.

@anitagraham could you remove it and squash your commits ?

**Comment by anitagraham** [comment]
I will make a couple more changes to the tests and squash.
I want to change the tests so that the elements and attributes being used will never be in the Loofah whitelist.

**Comment by anitagraham** [comment]
Also I thought set was needed because the Loofah whitelist is a set. Will see what happens when I remove it.

**Comment by anitagraham** [comment]
1 commit. 4 files. No 'require set'.

**Comment by sintro** [comment]
What about cross-model solution (Refinery::Blog::Post, and probably user-defined extensions)?

**Comment by bricesanchez** [comment]
@anitagraham Could rebase your commits ?

@sintro I'm not sure to understand your idea

**Comment by sintro** [comment]
@bricesanchez, as I can assume from fast reading of commits, all the changes are focusing on Refinery::Page sanitizing when presenting it's content. Meanwhile Refinery::Blog::Post will be untouched and admin will not be able to set tags and attributes whitelists for Posts content, am I right?

**Comment by bricesanchez** [comment]
@sintro The changes are focusing on the section presenter which is used to display blog post too : https://github.com/refinery/refinerycms-blog/blob/master/app/views/refinery/blog/posts/index.html.erb#L19

So `Refinery::Blog::Post` will be touched too and you will have to add missing elements/attributes in your Refinery::Page config.

**Comment by parndt** [comment]
What if you just have the blog extension but not the pages extension? Sounds like coupling.

**Comment by bricesanchez** [comment]
@parndt so there will be no sanitize

**Comment by parndt** [comment]
My main concern is unrelated to this change which is that content_page relies on a helper in the pages extension, which is not guaranteed to be there 😦

**Comment by bricesanchez** [comment]
You're right, in order to test coupling, we will probably need to extract all engines from the core?

**Comment by parndt** [comment]
@bricesanchez you can do this now: 

```
rails _4.2.6_ new blog_test_app
```

If you put just the blog in your project's Gemfile:

```ruby
gem "refinerycms-blog", git: "https://github.com/refinery/refinerycms-blog"
```

This will merely pull in `refinerycms-core` and `refinerycms-settings` based on the current gemspec.  Then you can run:

```
bundle install
rails g refinery:cms
```

And all the other normal set up tasks like migrations, and you should have a Refinery with _only_ the Blog, and this _should_ fail when you open its index page, but also much before that:

```
undefined method `translates' for #<Class:0x007fd4e4b6de18>
```

😦

**Comment by bricesanchez** [comment]
Is there a doc for this awesome thing?

**Comment by parndt** [comment]
@bricesanchez no, it's just been quietly supported for a couple of major versions 😄

**Comment by anitagraham** [comment]
@sintro If  you have a standard Refinery with Refinery-Blog extension, then the changes to the Pages config will also apply to the Blog posts as I think they both use the same code.  I will have to check that to be sure.

**Comment by anitagraham** [comment]
@parndt, @bricesanchez 
You are right. The handling of HTML content/page parts should be separated from Pages. 

Pages:  Title, Page parts, control parts (now called advanced settings).

I would really like to discuss this more - as a route to making pages/parts/controls/extensions more othogonal to each other.

**Comment by anitagraham** [comment]
PS. I don't understand the conflicts. If I follow the command line instructions nothing seems to change.

**Comment by bricesanchez** [comment]
Closed in favor of #3164

**Pull request closed without merging**